molecule/scenario.py

```diff
@@ -226,8 +232,24 @@ def _setup(self):
             os.makedirs(self.inventory_directory)
 
 
-def ephemeral_directory(path):
+def ephemeral_directory(path=None):
+    """
+    Returns temporary directory to be used by molecule. Molecule users should
+    not make any assumptions about its location, permissions or its content as
+    this may change in future release.
+    """
     d = os.getenv('MOLECULE_EPHEMERAL_DIRECTORY')
-    if d:
-        return os.path.join(tempfile.gettempdir(), d)
-    return os.path.join(tempfile.gettempdir(), path)
+    if not d:
+        # Darwin is the only known platform that returns user-isolated tempdirs
+        # so we do not need to make them unique to each user.
+        if platform.system() == 'Darwin':
+            d = tempfile.gettempdir()
+        else:
+            d = os.path.join(tempfile.gettempdir(), 'mol.' + getpass.getuser())
+    else:
+        d = os.path.abspath(d)
+    if path:
+        d = os.path.abspath(os.path.join(d, path))
+        if not os.path.isdir(d):
+            Path(d).mkdir(parents=True, exist_ok=True)
+    return d
```

Comment:
This function is definitely not safe.  ie: malicious users can pre-create paths to gain control over the data that is in here, possibly get molecule to overwrite important files, etc.

This problem might affect the older code as well but one new thing that makes this code less secure is the ```mkdir(parents=True``` which is present here.  If we remove the mkdir from this function, then it is at least, no less secure than the previous code.

Reply:
Why this was ignored, I have no idea.